modules/runtime/nuxeo-runtime-test/src/main/java/org/nuxeo/runtime/test/runner/FeaturesRunner.java

```diff
@@ -167,42 +179,108 @@ public URL getTargetTestResource(String name) throws IOException {
     }
 
     protected void afterMethodRun(final FrameworkMethod method, final Object test) {
-        loader.apply(Direction.FORWARD, holder -> holder.feature.afterMethodRun(FeaturesRunner.this, method, test));
+        apply("afterMethodRun", FORWARD, holder -> holder.feature.afterMethodRun(FeaturesRunner.this, method, test));
     }
 
     protected void afterRun() {
-        loader.apply(Direction.BACKWARD, holder -> holder.feature.afterRun(FeaturesRunner.this));
+        apply("afterRun", BACKWARD, holder -> holder.feature.afterRun(FeaturesRunner.this));
     }
 
     protected void start() {
-        loader.apply(Direction.FORWARD, holder -> holder.feature.start(FeaturesRunner.this));
+        apply("start", FORWARD, holder -> holder.feature.start(FeaturesRunner.this));
     }
 
     protected void stop() {
-        loader.apply(Direction.BACKWARD, holder -> holder.feature.stop(FeaturesRunner.this));
+        apply("stop", BACKWARD, holder -> holder.feature.stop(FeaturesRunner.this));
     }
 
     protected void beforeSetup(final FrameworkMethod method, final Object test) {
-        loader.apply(Direction.FORWARD, holder -> holder.feature.beforeSetup(FeaturesRunner.this, method, test));
+        apply("beforeSetup", FORWARD, holder -> holder.feature.beforeSetup(FeaturesRunner.this, method, test));
         injector.injectMembers(underTest);
     }
 
     protected void afterTeardown(final FrameworkMethod method, final Object test) {
-        loader.apply(Direction.BACKWARD, holder -> holder.feature.afterTeardown(FeaturesRunner.this, method, test));
+        apply("afterTeardown", BACKWARD, holder -> holder.feature.afterTeardown(FeaturesRunner.this, method, test));
+    }
+
+    /**
+     * @since 11.1
+     */
+    protected void apply(String id, Direction direction, ThrowableConsumer<FeaturesLoader.Holder, Exception> consumer) {
+        apply(id, direction == FORWARD ? loader.holders() : loader.reversedHolders(), consumer);
+    }
+
+    /**
+     * @since 11.1
+     */
+    protected void apply(String id, Collection<FeaturesLoader.Holder> holders,
+            ThrowableConsumer<FeaturesLoader.Holder, Exception> consumer) {
+        log.trace("Invoke: {} on features: {}", () -> id, () -> formatFeatures(holders));
+        Supplier<String> msg = () -> String.format("Error while invoking %s on features: %s", id,
+                formatFeatures(holders));
+        var errors = new ArrayList<Throwable>();
+        for (FeaturesLoader.Holder each : holders) {
+            try {
+                consumer.accept(each);
+            } catch (AssumptionViolatedException cause) {
+                log.debug("Test ignored by the feature: {}", each.type);
+                throw cause;
+            } catch (InterruptedException cause) {
+                Thread.currentThread().interrupt();
+                throw new AssertionError(msg.get(), cause);
+            } catch (Throwable cause) { // NOSONAR
```

Comment:
Could you also just rethrow immediately `Error`, like out of memory or stack overflow, instead of wrapping it?